### ISSUE TITLE
bug: "\n" in json answer from http are interpreted in bodyjson

### ISSUE THREAD

**Issue by JeGoi** [ovh/venom#493]
Hi,
We are using the http executor to add info in our api.
The answer contains "\n" in one variable (result.bodyjson.items.item0.cert):

**==Testing Code==** 
Executor
```
executor: pf_api_action
input:
  body: {}
  method: GET
  status_code: 200
  url: ""
  time_to_sleep: 5
steps:
- type: get_login_token

- type: http
  method: "{{.input.method}}"
  url: '{{.pfserver_webadmin_url}}/api/v1/{{.input.url}}'
  ignore_verify_ssl: true
  body: '{{.input.body}}'
  headers:
    "Authorization": "{{.result.token}}"
    "Content-Type": "application/json"
  assertions:
    - result.statuscode ShouldEqual "{{.input.status_code}}"
    - result.bodyjson.__type__ ShouldEqual Map
  vars:
    http_json:
      from: result.bodyjson

output:
  pf_api_action_: "{{.http_json}}" # E1
  pf_api_actiong_: '{{.http_json}}' # E2
  pf_api_action_bodyjson_: '{{.result.bodyjson}}' # E3
  pf_api_id_: "{{.result.bodyjson.items.items0.id}}" # E4
  pf_api_http_json_id_: "{{.http_json.items.items0.id}}" # E5
```

test_suite
```
name: testsuite test pf api pki output
testcases:
- name: pki add ca pf_api_action
  steps:
  - type: pf_api_action
    body: >-
      {
        "cn": "COUCOU23",
        "mail": "{{.wired_dot1x_eap_tls_manual.certs.ca.mail}}",
        "organisational_unit": "{{.wired_dot1x_eap_tls_manual.certs.organisational_unit}}",
        "organisation": "{{.wired_dot1x_eap_tls_manual.certs.organisation}}",
        "country": "{{.wired_dot1x_eap_tls_manual.certs.country}}",
        "state": "{{.wired_dot1x_eap_tls_manual.certs.state}}",
        "locality": "{{.wired_dot1x_eap_tls_manual.certs.locality}}",
        "key_type": "{{.wired_dot1x_eap_tls_manual.certs.key_type}}",
        "digest": "{{.wired_dot1x_eap_tls_manual.certs.digest}}",
        "key_usage": "",
        "extended_key_usage": "",
        "days": "{{.wired_dot1x_eap_tls_manual.certs.validity}}",
        "key_size": "{{.wired_dot1x_eap_tls_manual.certs.key_size}}",
        "ocsp_url": "{{.wired_dot1x_eap_tls_manual.certs.ocsp_url}}"
      }
    method: POST
    url: "pki/cas"
    info:
      - "{{.result.pf_api_actiong_}}" # S1 based on E2
      - "{{.result.pf_api_actiong_json.items.items0.id}}"  # S2 E2
      - "{{.result.pf_api_action_}}"  # S1 E1
      - "{{.result.pf_api_action_json}}"   # S2 E1
      - "{{.result.pf_api_action_json.contentType}}"   # S3 E1
      - "{{.result.pf_api_action_json.items.items0.id}}"   # S4 E1
      - "{{.result.pf_api_action_bodyjson_}}"   # S1 E3
      - "{{.result.pf_api_action_bodyjson_.items.items0.id}}"   # S2 E3
      - "{{.result.pf_api_action_bodyjson_json}}"   # S3 E3
      - "{{.result.pf_api_action_bodyjson_json.items.items0.id}}"   # S4 E3
      - "{{.result.pf_api_id_}}"   # S1 E4
      - "{{.result.pf_api_id_json}}"   # S2 E4
      - "{{.result.pf_api_http_json_id_}}"  # S1 E5
      - "{{.result.pf_api_http_json_id_json}}"   # S2 E5
```

**==LOG and INFO==** 
venom.log
```
Jan 24 15:13:02.538^[[36m [INFO] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [get_login_token] ^[[0mStep #0 content is: "{\"assertions\":[\"result.statuscode ShouldEqual \\\"200\\\"\"],\"body\":\"{\\n  \\\"username\\\": \\\"admin\\\",\\n  \\\"password\\\": \\\"secret\\\"\\n}      \",\"delay\":\"1\",\"headers\":{\"Content-Type\":\"application/json\"},\"ignore_verify_ssl\":true,\"method\":\"POST\",\"retry\":\"3\",\"type\":\"http\",\"url\":\"https://172.18.120.15:1443/api/v1/login\"}"
Jan 24 15:13:02.577^[[37m [DEBU] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [http] ^[[0mresult of runTestStepExecutor: {TimeSeconds:0.036041828 StatusCode:200 Request:{Method:POST URL:https://172.18.120.15:1443/api/v1/login Header:map[Content-Type:[application/json]] Body:{
  "username": "admin",
  "password": "secret"
}       Form:map[] PostForm:map[]} Body:{"token":"xx"} BodyJSON:map[token:xx] Headers:map[Content-Length:76 Content-Type:text/plain; charset=utf-8 Date:Mon, 24 Jan 2022 15:13:02 GMT Server:Caddy X-Frame-Options:SAMEORIGIN] Err:}
Jan 24 15:13:02.578^[[37m [DEBU] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [get_login_token] ^[[0mresult of runTestStepExecutor: map[:map[result:map[token:xx]] __len__:1 __len__json:1 __type__:Map json:map[result:map[token:xx]] result:map[token:xx] result.__len__:1 result.__len__json:1 result.__type__:Map result.token:xx result.tokenjson:7e5 resultjson:map[token:xx]]
Jan 24 15:13:02.737^[[36m [INFO] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [pf_api_action] ^[[0mStep #1 content is: "{\"assertions\":[\"result.statuscode ShouldEqual \\\"200\\\"\",\"result.bodyjson.__type__ ShouldEqual Map\"],\"body\":\"{ \\\"cn\\\": \\\"COUCOU24\\\", \\\"mail\\\": \\\"mailhog@example.lan\\\", \\\"organisational_unit\\\": \\\"PacketFence\\\", \\\"organisation\\\": \\\"Inverse\\\", \\\"country\\\": \\\"CA\\\", \\\"state\\\": \\\"Quebec\\\", \\\"locality\\\": \\\"Montreal\\\", \\\"key_type\\\": \\\"1\\\", \\\"digest\\\": \\\"4\\\", \\\"key_usage\\\": \\\"\\\", \\\"extended_key_usage\\\": \\\"\\\", \\\"days\\\": \\\"750\\\", \\\"key_size\\\": \\\"2048\\\", \\\"ocsp_url\\\": \\\"https://127.0.0.1:22225/api/v1/pki/ocsp\\\" }\",\"headers\":{\"Authorization\":\"xx\",\"Content-Type\":\"application/json\"},\"ignore_verify_ssl\":true,\"method\":\"POST\",\"type\":\"http\",\"url\":\"https://172.18.120.15:1443/api/v1/pki/cas\",\"vars\":{\"http_json\":{\"from\":\"result.bodyjson\"}}}"
Jan 24 15:13:03.402^[[37m [DEBU] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [http] ^[[0mresult of runTestStepExecutor: {TimeSeconds:0.661780283 StatusCode:200 Request:{Method:POST URL:https://172.18.120.15:1443/api/v1/pki/cas Header:map[Authorization:[xx] Content-Type:[application/json]] Body:{ "cn": "COUCOU24", "mail": "mailhog@example.lan", "organisational_unit": "PacketFence", "organisation": "Inverse", "country": "CA", "state": "Quebec", "locality": "Montreal", "key_type": "1", "digest": "4", "key_usage": "", "extended_key_usage": "", "days": "750", "key_size": "2048", "ocsp_url": "https://127.0.0.1:22225/api/v1/pki/ocsp" } Form:map[] PostForm:map[]} Body:{"status":0,"password":"","error":"","contentType":"application/json; charset=UTF-8","raw":null,"items":[{"ID":26,"CreatedAt":"0001-01-01T00:00:00Z","UpdatedAt":"0001-01-01T00:00:00Z","DeletedAt":null,"DB":{"Value":null,"Error":null,"RowsAffected":0},"Ctx":null,"cn":"COUCOU24","mail":"mailhog@example.lan","organisation":"Inverse","organisational_unit":"PacketFence","country":"CA","state":"Quebec","locality":"Montreal","key_type":"1","key_size":"2048","digest":"4","key_usage":"","extended_key_usage":"","days":"750","cert":"-----BEGIN CERTIFICATE-----\nMIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP\nMA0GA1UECBMGUXVlYmVjMREwDwYDVQQHEwhNb250cmVhbDEQMA4GA1UEChMHSW52\nZXJzZTEUMBIGA1UECxMLUGFja2V0RmVuY2UxETAPBgNVBAMTCENPVUNPVTI0MB4X\nDTIyMDEyNDE1MTMwM1oXDTI0MDIxMzE1MTMwM1owbDELMAkGA1UEBhMCQ0ExDzAN\nBgNVBAgTBlF1ZWJlYzERMA8GA1UEBxMITW9udHJlYWwxEDAOBgNVBAoTB0ludmVy\nc2UxFDASBgNVBAsTC1BhY2tldEZlbmNlMREwDwYDVQQDEwhDT1VDT1UyNDCCASIw\nDQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBANBhahwk//8o3PxSPAahsE+kV9KF\nxSnm2bVI8lBPPKb/gX2iaWhuNL7EjpLvFt3qkXQiL/Hk1csCEwke7eEA+fmKdfYq\nZOEY5kpNKsesPT8Il2kNItwV2sm4Qp5vu8vUrMIycAgoBIFOlK6rLrZTFmQAUQ/9\nb9KyxWNL4HlUBFAa6oZX+KfUIj0BSZm89m+aBOyABzgDH7hE9nuIioDEXpMHU3rB\n3OwqsKbSxWoc3X9kCc1jwIx0iKIBX+rPoTJ21JDXyTbXcqcaYOTjXlBmo4vYxijn\nnYWlNk/YTJATudql+ZmNKai1xXx0wgAEQWo8F/5s1UHY5FrUpQYbq4lpJTMCAwEA\nAaOBuTCBtjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSeIDsUEWmalnOuLHJi\nCtiV6KIMlzAfBgNVHSMEGDAWgBSeIDsUEWmalnOuLHJiCtiV6KIMlzBDBggrBgEF\nBQcBAQQ3MDUwMwYIKwYBBQUHMAGGJ2h0dHBzOi8vMTI3LjAuMC4xOjIyMjI1L2Fw\naS92MS9wa2kvb2NzcDAeBgNVHREEFzAVgRNtYWlsaG9nQGV4YW1wbGUubGFuMA0G\nCSqGSIb3DQEBCwUAA4IBAQBuPYb0CEKnHIXYqP1VdykedlkkLBuPksuV5tQZbRzo\nBiwfRz3/qpVqaxOVDGhhQIE1RhqaxmBCOtuBZFQznbR31avT9BCtwDrmjWQiUwf4\nHvG7eelD9wjjcX7+hDYqmeYhMnB81ZoK+G60zU2Lk4cNKxU6rCbd7oOSAldL5Xky\neNTOOYU4hBkErIwzF62bS7jVT4xOa8sgax8hpVAy82PVBLrlYRMPDOeA101a2Xhm\nsKcDk1ytgMrH+K/pcZbzOENRxYWmHTtp4Ccw5JmQg7VSeehoBb+6t0YzseZbUCdn\nKxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M\n-----END CERTIFICATE-----\n","ocsp_url":"https://127.0.0.1:22225/api/v1/pki/ocsp","SCEPAssociateProfile":"","Cloud":null}],"nextCursor":0,"prevCursor":0,"total_count":0,"serial":""}
 BodyJSON:map[contentType:application/json; charset=UTF-8 error: items:[map[Cloud:<nil> CreatedAt:0001-01-01T00:00:00Z Ctx:<nil> DB:map[Error:<nil> RowsAffected:0 Value:<nil>] DeletedAt:<nil> ID:26 SCEPAssociateProfile: UpdatedAt:0001-01-01T00:00:00Z cert:-----BEGIN CERTIFICATE-----
MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP
[...]
sKcDk1ytgMrH+K/pcZbzOENRxYWmHTtp4Ccw5JmQg7VSeehoBb+6t0YzseZbUCdn
KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M
-----END CERTIFICATE-----
 cn:COUCOU24 country:CA days:750 digest:4 extended_key_usage: key_size:2048 key_type:1 key_usage: locality:Montreal mail:mailhog@example.lan ocsp_url:https://127.0.0.1:22225/api/v1/pki/ocsp organisation:Inverse organisational_unit:PacketFence state:Quebec]] nextCursor:0 password: prevCursor:0 raw:<nil> serial: status:0 total_count:0] Headers:map[Content-Type:application/json Date:Mon, 24 Jan 2022 15:13:03 GMT Server:Caddy X-Frame-Options:SAMEORIGIN] Err:}
Jan 24 15:13:03.403^[[37m [DEBU] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [pf_api_action] ^[[0mProcessing http_json assignment
Jan 24 15:13:03.403^[[36m [INFO] [testsuite test pf api pki output] [pki-add-ca-pf_api_action] [pf_api_action] ^[[0mAssign 'http_json' value 'map[contentType:application/json; charset=UTF-8 error: items:[map[Cloud:<nil> CreatedAt:0001-01-01T00:00:00Z Ctx:<nil> DB:map[Error:<nil> RowsAffected:0 Value:<nil>] DeletedAt:<nil> ID:26 SCEPAssociateProfile: UpdatedAt:0001-01-01T00:00:00Z cert:-----BEGIN CERTIFICATE-----
MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP
[...]
sKcDk1ytgMrH+K/pcZbzOENRxYWmHTtp4Ccw5JmQg7VSeehoBb+6t0YzseZbUCdn
KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M
-----END CERTIFICATE-----
 cn:COUCOU24 country:CA days:750 digest:4 extended_key_usage: key_size:2048 key_type:1 key_usage: locality:Montreal mail:mailhog@example.lan ocsp_url:https://127.0.0.1:22225/api/v1/pki/ocsp organisation:Inverse organisational_unit:PacketFence state:Quebec]] nextCursor:0 password: prevCursor:0 raw:<nil> serial: status:0 total_count:0]'
```

Venom Info:
```
        • pki-add-ca-pf_api_action SUCCESS
          [info] {"contentType":"application/json; charset=UTF-8","error":"","items":[{"Cloud":null,"CreatedAt":"0001-01-01T00:00:00Z","Ctx":null,"DB":{"Error":null,"RowsAffected":0,"Value":null},"DeletedAt":null,"ID":26,"SCEPAssociateProfile":"","UpdatedAt":"0001-01-01T00:00:00Z","cert":"-----BEGIN CERTIFICATE-----
MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP
[...]
KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M
-----END CERTIFICATE-----
","cn":"COUCOU24","country":"CA","days":"750","digest":"4","extended_key_usage":"","key_size":"2048","key_type":"1","key_usage":"","locality":"Montreal","mail":"mailhog@example.lan","ocsp_url":"https://127.0.0.1:22225/api/v1/pki/ocsp","organisation":"Inverse","organisational_unit":"PacketFence","state":"Quebec"}],"nextCursor":0,"password":"","prevCurso
r":0,"raw":null,"serial":"","status":0,"total_count":0} 
          [info] {{.result.pf_api_actiong_json.items.items0.id}} 
          [info] {"contentType":"application/json; charset=UTF-8","error":"","items":[{"Cloud":null,"CreatedAt":"0001-01-01T00:00:00Z","Ctx":null,"DB":{"Error":null,"RowsAffected":0,"Value":null},"DeletedAt":null,"ID":26,"SCEPAssociateProfile":"","UpdatedAt":"0001-01-01T00:00:00Z","cert":"-----BEGIN CERTIFICATE-----
MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP
[...]
KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M
-----END CERTIFICATE-----
","cn":"COUCOU24","country":"CA","days":"750","digest":"4","extended_key_usage":"","key_size":"2048","key_type":"1","key_usage":"","locality":"Montreal","mail":"mailhog@example.lan","ocsp_url":"https://127.0.0.1:22225/api/v1/pki/ocsp","organisation":"Inverse","organisational_unit":"PacketFence","state":"Quebec"}],"nextCursor":0,"password":"","prevCursor":0,"raw":null,"serial":"","status":0,"total_count":0} 
          [info] {{.result.pf_api_action_json}} 
          [info] {{.result.pf_api_action_json.contentType}} 
          [info] {{.result.pf_api_action_json.items.items0.id}} 
          [info] {"contentType":"application/json; charset=UTF-8","error":"","items":[{"Cloud":null,"CreatedAt":"0001-01-01T00:00:00Z","Ctx":null,"DB":{"Error":null,"RowsAffected":0,"Value":null},"DeletedAt":null,"ID":26,"SCEPAssociateProfile":"","UpdatedAt":"0001-01-01T00:00:00Z","cert":"-----BEGIN CERTIFICATE-----
MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP
[...]
KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M
-----END CERTIFICATE-----
","cn":"COUCOU24","country":"CA","days":"750","digest":"4","extended_key_usage":"","key_size":"2048","key_type":"1","key_usage":"","locality":"Montreal","mail":"mailhog@example.lan","ocsp_url":"https://127.0.0.1:22225/api/v1/pki/ocsp","organisation":"Inverse","organisational_unit":"PacketFence","state":"Quebec"}],"nextCursor":0,"password":"","prevCursor":0,"raw":null,"serial":"","status":0,"total_count":0} 
          [info] {{.result.pf_api_action_bodyjson_.items.items0.id}} 
          [info] {{.result.pf_api_action_bodyjson_json}} 
          [info] {{.result.pf_api_action_bodyjson_json.items.items0.id}} 
          [info] 26 
          [info] 26 
          [info] 26 
          [info] 26 
```

**==ISSUE==** 
As long as we are in the executor, we are able to get the json content with:
```
  pf_api_id_: "{{.result.bodyjson.items.items0.id}}"
  pf_api_http_json_id_: "{{.http_json.items.items0.id}}"
```
BUT when we are out of the executor (var reassign before or not), using single or double cote on the variable, it looks impossible to get access to the variable inside the passing variable.

**==EXPECTED==** 
Having the same variables json access in the test suite and the executor to be able to do specific assertions for this request.

**==WORKAROUND==** 
Set the needed variables in the executor, and call it in the test suite, or other executor.
If the variable is not here, variable=nil

**==NOTES==** 
- to make it a bit more readable, I replace few lines with [...]
- Our json is a MAP not an Array
- variable with issue: "cert":"-----BEGIN CERTIFICATE----- **\n** MIIEDTCCAvWgAwIBAgIBGjANBgkqhkiG9w0BAQsFADBsMQswCQYDVQQGEwJDQTEP **\n** MA0GA1UECBMGUXVlYmVjMREwDwYDVQQHEwhNb250cmVhbDEQMA4GA1UEChMHSW52 **\n** ZXJzZTEUMBIGA1UECxMLUGFja2V0RmVuY2UxETAPBgNVBAMTCENPVUNPVTI0MB4X **\n** DTIyMDEyNDE1MTMwM1oXDTI0MDIxMzE1MTMwM1owbDELMAkGA1UEBhMCQ0ExDzAN **\n** BgNVBAgTBlF1ZWJlYzERMA8GA1UEBxMITW9udHJlYWwxEDAOBgNVBAoTB0ludmVy **\n** c2UxFDASBgNVBAsTC1BhY2tldEZlbmNlMREwDwYDVQQDEwhDT1VDT1UyNDCCASIw **\n** DQYJKoZIhvcNAQEBBQADggEPADCCAQoCggEBANBhahwk//8o3PxSPAahsE+kV9KF **\n** xSnm2bVI8lBPPKb/gX2iaWhuNL7EjpLvFt3qkXQiL/Hk1csCEwke7eEA+fmKdfYq **\n** ZOEY5kpNKsesPT8Il2kNItwV2sm4Qp5vu8vUrMIycAgoBIFOlK6rLrZTFmQAUQ/9 **\n** b9KyxWNL4HlUBFAa6oZX+KfUIj0BSZm89m+aBOyABzgDH7hE9nuIioDEXpMHU3rB **\n** 3OwqsKbSxWoc3X9kCc1jwIx0iKIBX+rPoTJ21JDXyTbXcqcaYOTjXlBmo4vYxijn **\n** nYWlNk/YTJATudql+ZmNKai1xXx0wgAEQWo8F/5s1UHY5FrUpQYbq4lpJTMCAwEA **\n** AaOBuTCBtjAPBgNVHRMBAf8EBTADAQH/MB0GA1UdDgQWBBSeIDsUEWmalnOuLHJi **\n** CtiV6KIMlzAfBgNVHSMEGDAWgBSeIDsUEWmalnOuLHJiCtiV6KIMlzBDBggrBgEF **\n** BQcBAQQ3MDUwMwYIKwYBBQUHMAGGJ2h0dHBzOi8vMTI3LjAuMC4xOjIyMjI1L2Fw **\n** aS92MS9wa2kvb2NzcDAeBgNVHREEFzAVgRNtYWlsaG9nQGV4YW1wbGUubGFuMA0G **\n** CSqGSIb3DQEBCwUAA4IBAQBuPYb0CEKnHIXYqP1VdykedlkkLBuPksuV5tQZbRzo **\n** BiwfRz3/qpVqaxOVDGhhQIE1RhqaxmBCOtuBZFQznbR31avT9BCtwDrmjWQiUwf4 **\n** HvG7eelD9wjjcX7+hDYqmeYhMnB81ZoK+G60zU2Lk4cNKxU6rCbd7oOSAldL5Xky **\n** eNTOOYU4hBkErIwzF62bS7jVT4xOa8sgax8hpVAy82PVBLrlYRMPDOeA101a2Xhm **\n** sKcDk1ytgMrH+K/pcZbzOENRxYWmHTtp4Ccw5JmQg7VSeehoBb+6t0YzseZbUCdn **\n** KxkxvqhRe6UcQserSiZ+aSdohotNPyUZ3+OyrMZiw75M **\n** -----END CERTIFICATE----- **\n** "

**Comment by JeGoi** [comment]
similar to: https://github.com/ovh/venom/issues/493

**Comment by yesnault** [comment]
Hello. It's probably fixed in the latest 1.1 version. Can your retry please?
If it's not fixed, feel free to reopen this ticket with a ready-to-run testsuite / user executor to reproduce the issue please. 

Thank you
